src/main/java/com/icthh/xm/ms/entity/validator/EventDataTypeKeyValidator.java

```diff
@@ -0,0 +1,45 @@
+package com.icthh.xm.ms.entity.validator;
+
+import static org.apache.commons.lang3.StringUtils.isBlank;
+
+import com.icthh.xm.commons.exceptions.BusinessException;
+import com.icthh.xm.commons.exceptions.EntityNotFoundException;
+import com.icthh.xm.ms.entity.domain.Event;
+import com.icthh.xm.ms.entity.domain.XmEntity;
+import com.icthh.xm.ms.entity.domain.spec.EventSpec;
+import com.icthh.xm.ms.entity.service.XmEntitySpecService;
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@RequiredArgsConstructor
+public class EventDataTypeKeyValidator implements ConstraintValidator<EventDataTypeKey, Event> {
+
+    private final XmEntitySpecService xmEntitySpecService;
+
+    @Override
+    public boolean isValid(Event event, ConstraintValidatorContext constraintValidatorContext) {
+        XmEntity eventDataRef = event.getEventDataRef();
+        if (eventDataRef == null) {
+            return true;
+        }
+
+        String eventTypeKey = event.getTypeKey();
+        EventSpec eventSpec = xmEntitySpecService.findEvent(eventTypeKey)
+            .orElseThrow(() -> new EntityNotFoundException("Event specification not found by key: " + eventTypeKey));
```

Comment:
here propose also to throw BusnessException

Reply:
already discussed in https://github.com/xm-online/xm-ms-entity/pull/217#discussion_r491983113